src/server/pfs/s3/s3_test.go

```diff
@@ -10,50 +10,39 @@ import (
 	"fmt"
 	"io"
 	"io/ioutil"
-	"net/http"
 	"os"
 	"path/filepath"
 	"sort"
 	"strings"
+	"sync"
 	"testing"
 	"time"
 
 	minio "github.com/minio/minio-go"
 	"github.com/pachyderm/pachyderm/src/client"
 	"github.com/pachyderm/pachyderm/src/client/pkg/require"
-	"github.com/pachyderm/pachyderm/src/server/pfs/server"
-	"github.com/pachyderm/pachyderm/src/server/pkg/backoff"
 	tu "github.com/pachyderm/pachyderm/src/server/pkg/testutil"
 )
 
-func serve(t *testing.T) (*http.Server, *client.APIClient, *minio.Client) {
-	t.Helper()
+var (
+	pachClient *client.APIClient
+	minioClient *minio.Client
+	clientOnce sync.Once
+)
 
-	port := tu.UniquePort()
-	pc := server.GetPachClient(t)
-	srv := Server(pc, port)
+func clients(t testing.TB) (*client.APIClient, *minio.Client) {
+	t.Helper()
 
-	go func() {
-		if err := srv.ListenAndServe(); err != http.ErrServerClosed {
```

Comment:
If I'm understanding it correctly, the s3 API is always running in the server - the command-line `s3gateway` would just run a local proxy?  I'm trying to understand why we no longer need to tell the server to listen here.

Reply:
There's an instance of the s3gateway now running on pachd. Because these have been switched over to integration tests, we no longer need to spawn a per-test-instance s3gateway.